runtime/datastructures/src/decls.rs

```diff
@@ -69,6 +101,86 @@ impl Hash for KElem {
   }
 }
 
+fn hash_map_compare<K,V>(a: &HashMap<K, V>, b: &HashMap<K, V>) -> Ordering
+where K: Ord + Hash + Clone, V: Ord + Clone {
+    let mut a_keys : BinaryHeap<_> = a.keys().collect();
+    let mut b_keys : BinaryHeap<_> = b.keys().collect();
+    while !(a_keys.is_empty() || b_keys.is_empty()) {
+        let (a_top, b_top) = (a_keys.pop(), b_keys.pop());
+        if a_top > b_top {
+            return Ordering::Greater;
+        }
+        if b_top > a_top {
+            return Ordering::Less;
+        }
+        let (a_top_elem, b_top_elem) = (a.get(a_top.unwrap()), b.get(b_top.unwrap()));
+        if a_top_elem > b_top_elem {
+            return Ordering::Greater;
+        }
+        if b_top_elem > a_top_elem {
+            return Ordering::Less;
+        }
+    }
+    if a_keys.is_empty() {
+        if b_keys.is_empty() {
+            return Ordering::Equal;
+        }
+        return Ordering::Less;
+    }
+    return Ordering::Greater;
+}
+
+fn hash_set_compare<K>(a: &HashSet<K>, b: &HashSet<K>) -> Ordering
+where K: Ord + Hash + Clone {
```

Comment:
Same here, this can also be deleted

Reply:
Going to resolve this while leaving the other open, whatever we do will be applied to both :)